src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2001,21 +2005,19 @@ RunDistributedExecution(DistributedExecution *execution)
 {
 	WaitEvent *events = NULL;
 
-	AssignTasksToConnections(execution);
+	AssignTasksToSameConnectionsIfNecessary(execution);
```

Comment:
I think SameConnectionsIfNecessary is not a comprehensive description of what this function does. Maybe just keep it the same?

Reply:
I think the problem comes from the fact that the function does many things. It assigns tasks to the same connections that were used previously for the placements, if not then it assigns it to a workerPool so that later it can be executed on any available connection right? `AssignTasksToConnections` doesnt seem to reflect that in my opinion. Because when this function returns, some of the tasks are not even assigned to connections but will be assigned later. Maybe this should be `AssignTasksToConnectionsOrWorkerPool` ?  This is what the functions does I guess

Reply:
@marcocitus what do you think?

Reply:
AssignTasksToConnectionsOrWorkerPool sounds good

Reply:
updated